pkg/controller/operconfig/migration.go

```diff
@@ -97,3 +112,313 @@ func convertEgressFirewallToEgressNetworkPolicy(ctx context.Context, client cnoc
 	}
 	return nil
 }
+
+func convertSdnEgressIpToOvnEgressIp(ctx context.Context, client cnoclient.Client) error {
+
+	// 1. query for hostsubnets and netnamespaces
+	hsnList, err := client.Default().Dynamic().Resource(gvrHostSubnet).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	nnsList, err := client.Default().Dynamic().Resource(gvrNetnamespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	// 2. iterate through hostsubnets
+	//    - any with egressIP configured will cause update to node label "egress-assignable"
+	hostSubnetFound := false
+	for _, hsn := range hsnList.Items {
+		if hostSubnetHasEgressIpConfigManual(hsn) {
+			// manual config detected, quit with error
+			return fmt.Errorf("Detected manually configured egressIP; only automatic configurations may be migrated.")
+		} else if hostSubnetHasEgressIpConfigAutomatic(hsn) {
+			hostSubnetFound = true
+			// automatic config detected, translating to ovn
+			// - mark node from hostsubnet with annotation as follows:
+			//    k8s.ovn.org/egress-assignable: ""
+			hostStr := fmt.Sprintf("%v", hsn.Object["host"])
+			nodeObj, err := client.Default().Kubernetes().CoreV1().Nodes().Get(ctx, hostStr, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			nodeObj.Labels["k8s.ovn.org/egress-assignable"] = ""
+			_, err = client.Default().Kubernetes().CoreV1().Nodes().Update(ctx, nodeObj, metav1.UpdateOptions{})
+			if err != nil {
+				return err
+			}
+
+			// now remove egressIP config from hostsubnet
+			hsn.Object["egressCIDRs"] = nil
+			hsn.Object["egressIPs"] = nil
+			_, err = client.Default().Dynamic().Resource(gvrHostSubnet).Update(ctx, &hsn, metav1.UpdateOptions{})
+			if err != nil {
+				return err
+			}
+		}
+
+	}
+	if !hostSubnetFound {
+		return fmt.Errorf("did not find a hostsubnet object with egressIP configured")
+	}
+
+	// 3. iterate through netnamespaces
+	//    - any with egressIP configured will cause an egressIP ovn resource to be created via k8s api
+	//    - a corresponding namespace label will be added to match the egressIP resource's namespace selector field
+	for i, nns := range nnsList.Items {
+		if netNamespaceHasEgressIpConfig(nns) {
+			// config detected, translating to ovn
+			// - delete cloudprivateipconfig if it exists
+			// - create egressip custom resource for ovn
+
+			for _, egressIp := range nns.Object["egressIPs"].([]interface{}) {
+				egressIpStr := fmt.Sprintf("%v", egressIp)
+				err = client.Default().Dynamic().Resource(gvrCloudPrivateIPConfig).Delete(ctx, egressIpStr, metav1.DeleteOptions{})
```

Comment:
We can do this check in the rollback code. However, here this loop is redundant and I plan on removing it entirely. 

If there is any cloudprivateipconfig, it shall be removed by SDN automatically. This is because code previously deletes any hostsubnet egressIP config. As a result SDN will clean up all CloudPrivateIpConfig attached to the hostsubnet egressIP configs.